### PR TITLE
Chore: Rewrite Location modal to React

### PR DESCRIPTION
**Pull request by gabriellsh**

### CODE REVIEW

File: apps/meteor/app/ui-message/client/messageBox/messageBoxActions.js [hunk 1]
```diff
@@ -69,123 +69,12 @@ messageBox.actions.add('Add_files_from', 'Computer', {
...
-	}
-
-	const position = await getGeolocationPosition();
-	return position;
-};
-
 messageBox.actions.add('Share', 'My_location', {
 	id: 'share-location',
 	icon: 'map-pin',
 	condition: () => canGetGeolocation.get(),
```
Comment (dougfabris) [inline apps/meteor/app/ui-message/client/messageBox/messageBoxActions.js:75]: this condition blocks the user to access the modal where can get the information about allowing the browser to share location

File: apps/meteor/client/views/room/ShareLocation/ShareLocationModal.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+				message: {
+					rid,
+					tmid,
+					location: {
+						type: 'Point',
+						coordinates: [positionData.coords.longitude, positionData.coords.latitude],
+					},
+				},
+			});
+		} catch (error: any) {
```
Comment (dougfabris) [inline apps/meteor/client/views/room/ShareLocation/ShareLocationModal.tsx:47]:
```suggestion
		} catch (error) {
```

File: apps/meteor/client/views/room/ShareLocation/ShareLocationModal.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+			dispatchToast({ type: 'error', message: error });
```
Comment (dougfabris) [inline apps/meteor/client/views/room/ShareLocation/ShareLocationModal.tsx:48]:
```suggestion
			dispatchToast({ type: 'error', message: error as Error });
```

File: apps/meteor/client/views/room/ShareLocation/ShareLocationModal.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+			onClose();
+		}
+	};
+
+	const onConfirmRequestLocation = async (): Promise<void> => {
+		try {
+			const position = await getGeolocationPosition();
+			queryClient.setQueryData(['geolocationPosition', 'granted'], position);
+			queryClient.setQueryData('geolocationPermission', 'granted');
+		} catch (e: any) {
```
Comment (dougfabris) [inline apps/meteor/client/views/room/ShareLocation/ShareLocationModal.tsx:59]:
```suggestion
		} catch (e) {
```

File: apps/meteor/client/views/room/ShareLocation/ShareLocationModal.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+			queryClient.setQueryData('geolocationPermission', () => getGeolocationPermission);
+		}
+	};
+
+	if (permissionLoading || permissionData === 'prompt') {
+		return (
+			<GenericModal
+				title={t('You_will_be_asked_for_permissions')}
```
Comment (dougfabris) [inline apps/meteor/client/views/room/ShareLocation/ShareLocationModal.tsx:67]: I think we can improve the modal content here by adding a generic title like 'Permission Required' and a text explaining that in order to share location it needs the browser permission action.

**Review by dougfabris — CHANGES_REQUESTED** [review]

File: apps/meteor/app/ui-message/client/messageBox/messageBoxActions.js (hunk 1 above)
Comment (gabriellsh) [reply apps/meteor/app/ui-message/client/messageBox/messageBoxActions.js:75]: As we discussed, we don't currently have a fallback for missing support on the position api, so we'll keep not showing the location button when not supported.

**Review by dougfabris — APPROVED** [review]

**Pull request merged**